test/compare.cpp

```diff
@@ -138,6 +138,24 @@ BOOST_AUTO_TEST_CASE(extract_and_comparator_then)
     BOOST_TEST((comp(S{1, "foo", 3.14}, S{1, "foo", 3.14}) == 0));
   }
 
+  // using lambda
+  {
+    const yk::compare::comparator auto comp =
+        extract(&S::id) | [](const S& s) { return s.name; } | [](const auto& s) { return s.height; };
```

Comment:
In order to *prove* this works, we need to test the rvalues are preserved and not destroyed on the comparison. We need a custom type for tracking lifetime (i.e. counts up on constructor, counts down on destructor). Existing types such as `int` and `std::string` can not prove that they are not destroyed; since the actual behavior of UB can not be determinably observed as long as we are using non-trackable types.

Reply:
Even after the pair programming session on this morning, I'm not sure how we can test the lifetime expiration. We have done it on a meta-programming level, but we still need to test it also on a runtime level.

Reply:
ASan was not working, at least.